File: src/_03_BehavioralPattern/_03_01_ChainOfResponsibilities/after/PrintRequestHandler.java
package _03_BehavioralPattern._03_01_ChainOfResponsibilities.after;

public class PrintRequestHandler extends RequestHandler {

  public PrintRequestHandler(RequestHandler nextHandler) {
    super(nextHandler);
  }

  @Override
  public void handler(Request request) {
    super.handler(request);
    System.out.println(request.getBody());
  }
  
}

File: src/_03_BehavioralPattern/_03_01_ChainOfResponsibilities/PrintRequestHandler.js
import RequestHandler from "./RequestHandler.js";

/**
 * @typedef { import("./Request").default } Request
 */

export default class PrintRequestHandler extends RequestHandler {
  /** @param { RequestHandler } nextHandler */
  constructor(nextHandler) {
    super(nextHandler);
  }

  /**
   * @override
   * @param { Request } request 
   */
  handler(request) {
    super.handler(request);
    console.log(request.getBody());
  }
}